rs/orchestrator/src/upgrade.rs

```diff
@@ -859,6 +859,10 @@ fn report_master_public_key_changed_metric(
     path: PathBuf,
     metrics: &OrchestratorMetrics,
 ) -> OrchestratorResult<()> {
+    // If the file doesn't exist then there is nothing to report.
+    if !path.try_exists().unwrap_or(true) {
```

Comment:
I'd personally find it easier to read if we returned early here in case of errors:
```suggestion
    if !path.try_exists().map_err(OrchestratorError::key_monitoring_error)? {
```